### ISSUE TITLE
Write article: What every C# developer should know about Starcounter

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#2857]
From #2855 

> I think our best shot at this is to be clear and precise in documenting it. We need some kind of overview article like What every C# developer should know about Starcounter that covers things like how the proxy objects work, how it relates to cloning/copying, how constructors and field intializers work (or doesn't work), how object equality is achieved, etc. And keep the focus on how it relates to the POCO experience in "standard" C# development.

And we even have a volunteer contributor - @mtornwall! From the same issue:

> That would be great to have, particularly upon first contact with Starcounter. I'd be happy to contribute tales of some of my own mistakes and misconceptions.

@warpech, do we have some good place for such an article over on [starcounter.io](http://starcounter.io/) maybe?

**Comment by warpech** [comment]
Definitely we should find a fit! 

So far, we have moved the material about Puppet web apps, RESTful web apps, Network and Typed JSON from GitHub wiki to starcounter.io

Now we need to create (or move) material regarding weaving with your help (and database stuff with @k-rus's help). Let's sit together on this.

Related: https://github.com/Starcounter/Starcounter/issues/1572

**Comment by per-samuelsson** [comment]
Off-topic:
@dan31, still struggle w/ not getting notifications; didn't get @warpech one from hours ago. And I have made the switch to personal gmail-account. Any news?

On-topic:
Reminder to include in the article: how *readonly* fields are bound.

**Comment by dan31** [comment]
@per-samuelsson, so you don't even get them to your gmail account?

**Comment by per-samuelsson** [comment]
> @per-samuelsson, so you don't even get them to your gmail account?

Nope. Some, but not all of them. Randomly. Just like before. I got this one you just wrote, but not the one posted by @warpech [here](https://github.com/Starcounter/Starcounter/issues/2857#issuecomment-135368073).

**Comment by mtornwall** [comment]
Not sure if this is entirely on topic, but I think there needs to be very clear descriptions of the isolation levels offered by Starcounter transactions.

As I understand it you get read-committed with long-running transactions, and snapshot isolation otherwise?

**Comment by dan31** [comment]
Yes, and you have an implicit read-only transaction in circumstances shortly touched there: http://starcounter.io/guides/transactions/running-background-jobs/. We will try eventually to explain the whole structure in a simple way. We also work on refining the transactions structure (and on naming) for 3.x. Warmly welcome is your current feedback (we can take that offline, @mtornwall, just fetch me some time at the office :)

**Comment by per-samuelsson** [comment]
> Not sure if this is entirely on topic, but I think there needs to be very clear descriptions of the isolation levels offered by Starcounter transactions.

Agree on both:

a) its not on topic (because this issue was about an article what works different when coding C# basic concepts compared to doing the same thing inside hosted Starcounter), and

b) we really need a throughout description of all the ins-and-outs of transactions and how they work in Starcounter, centered around common use-cases.

**Comment by mtornwall** [comment]
@per-samuelsson Well, I thought it was a little bit on topic in the sense that the workings of transactions certainly affect how you can, and should, write your code (particularly with regards to side effects etc). But I do concede that it probably belongs elsewhere. :-)

The following is hopefully a bit more on topic. Most C# developers are accustomed to being able to attach a debugger to a running process. A developer new to Starcounter may not realize that they need to attach to the code host. I suggest providing a short section outlining the following:

  1. The general procedure for debugging a running Starcounter application (attaching debugger to code host).
  2. How the PID of the code host can be found, if you have multiple databases.

**Comment by per-samuelsson** [comment]
> Most C# developers are accustomed to being able to attach a debugger to a running process. A developer new to Starcounter may not realize that they need to attach to the code host. I suggest providing a short section outlining the following [...]

Totally agree. Very good point.

> How the PID of the code host can be found, if you have multiple databases.

Yep, that should be part of such documentation. It's part of the domain model exposed by the admin server already, I don't know why the web administrator interface doesn't show it next to the application.

If someone didn't know, and/or need it now, `staradmin list db` will reveal it.

```
staradmin list db

a_b
  Running         False
  Host process ID n/a
  Applications    n/a

default
  Running         True
  Host process ID 7520
  Applications    1

```
